sklearn/ensemble/gradient_boosting.py

```diff
@@ -1836,3 +1843,353 @@ def apply(self, X):
         leaves = super(GradientBoostingRegressor, self).apply(X)
         leaves = leaves.reshape(X.shape[0], self.estimators_.shape[0])
         return leaves
+
+
+class GradientBoostingClassifierCV(BaseEstimator):
+    """ Gradient Boosting Classification with Cross Validation
+
+    This class keeps adding estimators to the underlying GB estimator till
```

Comment:
I would do something simpler. Fit n_estimators for each folds and select the best number of estimators using the validation folds. From that point, I would add other stopping criterions.